mantidimaging/core/io/saver.py

```diff
@@ -139,12 +140,21 @@ def save(images: Images,
 
         with progress:
             for idx in range(num_images):
-                write_func(data[idx, :, :], names[idx], overwrite_all)
+                # Overwrite images with the copy that has been rescaled.
+                if pixel_depth == "int16":
+                    int16_size = 65536
+                    write_func(rescale_single_image(images.data[idx], images.data.min(), images.data.max(),
+                                                    int16_size - 1), names[idx], overwrite_all)
+                else:
+                    write_func(data[idx, :, :], names[idx], overwrite_all)
```

Comment:
```suggestion
                # Overwrite images with the copy that has been rescaled.
                if pixel_depth == "int16":
                    image = rescale_single_image(np.copy(images.data[idx]), min_in, max_in, int16_max_val)
                else:
                    image = images.data[idx]
                write_func(image, names[idx], overwrite_all)
```

And I can't add this to the suggestion, but outside the for loop:

```
            int16_max_val = 65535
            min_in = images.data.min()
            max_in = images.data.max()
```

Reply:
I put them elsewhere in the file :+1: